### DISCUSSION TITLE
Search index generation query

### DISCUSSION THREAD

**Discussion by tigattack** [komoot/photon#847] [Q&A]
I see the README mentions that the search index is updated weekly and thankfully provided by [GraphHopper](https://www.graphhopper.com/) with the support of [lonvia](https://github.com/lonvia).

Is there a public repository containing the source for the search index generation process? I've not been able to find it.

I have a feature request and would be interested in implementing it myself & opening a PR. I'd like to include a JSON and/or text file in each directory containing one or more bzip-compressed datasets to provide the decompressed size of the datasets.

To be clear, I'm referring to the search indexes available at https://download1.graphhopper.com/public/

**Comment by lonvia** [comment]
There is no public repo for the deployment on https://download1.graphhopper.com/public/.

A textfile with sizes seems to be a rather niche feature request. The compression ratio is somewhere around 2.5 for the database. So a rule of thumb that says you need about 3 times as much free space as the compressed file is large should be sufficient.

**Reply by tigattack** [reply]
It's definitely rather niche. My thinking was if I implemented it myself perhaps the maintenance team would be more open to it, given there's little downside, and it could be useful to a number of people (granted, though, it is hard to gauge that). However, since the source isn't public, I will leave it up to the team to decide if it's worth implementing.

Personally, my use case is an Ansible role which includes [functionality to deploy Photon](https://github.com/tigattack/ansible-role-dawarich/blob/main/tasks/deploy_photon.yml) and attempts to determine if there is sufficient free space for the DB. Currently, it simply assumes a 2.5x compression ratio. My motivation is I would much prefer to work with exact figures rather than estimates, but I'm glad to hear my "guesstimate" aligns with your suggestion.
